pegen/c_generator.py

```diff
@@ -477,8 +481,14 @@ def join_conditions(self, keyword: str, node: Any, names: List[str]) -> None:
                 self.visit(item, names=names)
         self.print(")")
 
-    def emit_action(self, node: Alt) -> None:
+    def emit_action(self, node: Alt, generated: bool) -> None:
         self.print(f"res = {node.action};")
+        if not generated:
```

Comment:
I've forgotten what the `generated` attribute on a node stands for. When do we *not* generate this? It *looks* like it's skipped inside actions for `_tmp_` nodes (but it's generated for `_loop` nodes). But *why* do we skip it for `_tmp_` actions?

Reply:
> I've forgotten what the `generated` attribute on a node stands for.

We are setting that on this PR for the first time and is supposed to be `True` for all artificial rules, so I think that if this is not `True` for `_tmp_` actions sounds like a bug.

Reply:
I think the reason I wanted to skip artificial rules is that those cannot have actions that fail, they just propagate downstream results so checking for errors there may hurt performance without gain. I may be wrong or missing something, but I think that was my original intent.

Reply:
That almost makes sense, except some sub-rules *do* have actions. Maybe this is premature optimization?

E.g. check out `_tmp_121_rule` which has the `Py_comprehension` action from this fragment in simpy:
```
for_if_clauses[asdl_seq*]:
    | a=(y=[ASYNC] 'for' a=star_targets 'in' b=disjunction c=('if' z=disjunction { z })*
        { _Py_comprehension(a, b, c, (y == NULL) ? 0 : 1, p->arena) })+ { a }
```
This is called from `_loop1_25_rule`, which is called from `for_if_clauses_rule`. I think it may propagate too late.